kubetest/azure.go

```diff
@@ -0,0 +1,520 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/url"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pelletier/go-toml"
+
+	"github.com/Azure/azure-storage-blob-go/2016-05-31/azblob"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/satori/go.uuid"
+)
+
+var (
+	// azure specific flags
+	acsResourceName        = flag.String("acsengine-resource-name", "", "Azure Resource Name")
+	acsResourceGroupName   = flag.String("acsengine-resourcegroup-name", "", "Azure Resource Group Name")
+	acsLocation            = flag.String("acsengine-location", "westus2", "Azure ACS location")
+	acsMasterVmSize        = flag.String("acsengine-mastervmsize", "Standard_D2s_v3", "Azure Master VM size")
+	acsAgentVmSize         = flag.String("acsengine-agentvmsize", "Standard_D2s_v3", "Azure Agent VM size")
+	acsAdminUsername       = flag.String("acsengine-admin-username", "", "Admin username")
+	acsAdminPassword       = flag.String("acsengine-admin-password", "", "Admin password")
+	acsAgentPoolCount      = flag.Int("acsengine-agentpoolcount", 2, "Azure Agent Pool Count")
+	acsAgentOSType         = flag.String("acsengine-agentOSType", "Windows", "OS Type of Agent Nodes. Options: Windows|Linux")
+	acsTemplatePath        = flag.String("acsengine-template", "", "Azure Template Name")
+	acsDnsPrefix           = flag.String("acsengine-dnsprefix", "", "Azure K8s Master DNS Prefix")
+	acsEngineURL           = flag.String("acsengine-download-url", "", "Download URL for ACS engine")
+	acsEngineMD5           = flag.String("acsengine-md5-sum", "", "Checksum for acs engine download")
+	acsSSHPublicKeyPath    = flag.String("acsengine-public-key", "", "Path to SSH Public Key")
+	acsWinBinariesURL      = flag.String("acsengine-win-binaries-url", "", "Path to get the zip file containing kubelet and kubeproxy binaries for Windows")
+	acsHyperKubeURL        = flag.String("acsengine-hyperkube-url", "", "Path to get the kyberkube image for the deployment")
+	acsCredentialsFile     = flag.String("acsengine-creds", "", "Path to credential file for Azure")
+	acsOrchestratorRelease = flag.String("acsengine-orchestratorRelease", "1.11", "Orchestrator Profile for acs-engine")
+	acsWinZipBuildScript   = flag.String("acsengine-winZipBuildScript", "https://raw.githubusercontent.com/Azure/acs-engine/master/scripts/build-windows-k8s.sh", "Build script to create custom zip containing win binaries for acs-engine")
```

Comment:
do we expect to need this for example as a flag? we seem to have a lot more flags

Reply:
Fair enough 👍I think at some point unrelated to this PR we will need to revisit kubetest and come up with a better way to organize these for one, prefixing the flags here seems like a pretty solid stopgap. Thanks for that